components/partials/header.js

```diff
@@ -2,81 +2,109 @@ import Image from "next/image";
 import Logo from "../../assets/images/logo-header.svg";
 import Link from "next/link";
 import { useRouter } from "next/router";
+import React, { useState, useEffect } from "react";
 
 export default function Header() {
   const router = useRouter();
+  const [showHeader, setShowHeader] = useState(true);
+  var [lastScrollPos, setLastScrollPos] = useState(0);
+
+  useEffect(() => {
+    const handleScroll = () => {
+      const currentScrollPos = window.pageYOffset;
+      if (currentScrollPos > lastScrollPos && showHeader) {
+        setShowHeader(false);
+      } else if (currentScrollPos < lastScrollPos && !showHeader) {
+        setShowHeader(true);
```

Comment:
Can set both conditions in one like `setShowHeader(currentScrollPos < lastScrollPos && !showHeader)`
No need to add `if else if`